.circleci/config.yml

```diff
@@ -284,6 +284,24 @@ defaults:
             name: Install build dependencies
             command: ./.circleci/osx_install_dependencies.sh
 
+  - steps_install_python_dependencies: &steps_install_python_dependencies
+      steps:
+        - run:
+            name: Install python dependencies
+            command: |
+                sudo apt update
+                sudo apt install python3-pip --assume-yes --no-install-recommends
+                python3 -m pip install requests --user
+
+  - steps_install_foundry: &steps_install_foundry
+      steps:
+        - run:
+            name: Install foundry
+            command: |
+                # Note: Currently Foundry only has nightly builds and there is no release checksum
+                wget https://github.com/foundry-rs/foundry/releases/download/nightly/foundry_nightly_linux_amd64.tar.gz --output-document /tmp/foundry.tar.gz
+                sudo tar --extract --gzip --file /tmp/foundry.tar.gz --directory /usr/local/bin
```

Comment:
Since it's neatly limited to a single dir, you could easily use `save_cache`/`restore_cache` to avoid having to download it again for each job.